### PR TITLE
Skip already built check if archive_site is empty

### PR DESCRIPTION
**Pull request by l2dy**
Hack for https://trac.macports.org/ticket/56110. e.g. Use `./mpbb list-subports --archive-site= git` in mpbot.

Hint for myself: exclude reasons are printed to stderr, so this is the only hack needed.

### CODE REVIEW

File: mpbb-list-subports [hunk 1]
```diff
@@ -48,7 +48,7 @@ print-subports() {
         # $thisdir is set in mpbb
         # shellcheck disable=SC2154
         archive_path=$("${tclsh}" "${thisdir}/tools/archive-path.tcl" "${port}")
-        if [[ -f "${archive_path}" ]]; then
+        if [[ -f "${archive_path}" && -n "${archive_site}" ]]; then
```
Comment (raimue) [inline mpbb-list-subports]: Couldn't we also skip `archive-path.tcl` if `archive_site` is empty?

Comment (l2dy) [reply mpbb-list-subports]: Yes, but I'd like to make a rather small change. Feel free to push a better version instead.

Comment (raimue) [reply mpbb-list-subports]: I pushed a new commit to the pull request branch. That seemed easier than trying to explain what I meant, which was only to move the conditional.

**Review by mojca — APPROVED** [review]

**Review by yan12125 — APPROVED** [review]

**Comment by l2dy** [comment]
Merged as 868f68814515c2af4dc4c4921b489b7ded410e99.

**Pull request closed without merging**